ConsoleApp6/PaymentServices.cs

```diff
@@ -15,35 +18,53 @@ public PaymentServices(Products p)
             this.UPC = p.UPC;
             this.Tax = Products.Tax;
             this.Discount = Products.Discount;
+            this.specialUPC = Products.specialUPC;
+            this.upcDiscount = Products.UPC_Discount;
          
         }
-        public double addTax()
+        double AddTax()
         {
 
-            double priceWithTax = Price + (Price * Tax / 100);
+            double priceWithTax = Price+(Price * Tax / 100);
             return Math.Round(priceWithTax, 2);
         }
 
-        public double createDiscount()
+        double CreateUniDiscount()
         {
-            double pricewithDiscount = addTax() - (Price * Discount / 100);
-            return Math.Round(pricewithDiscount, 2);
+            double amountofDiscount =Price * Discount / 100;
+            //check if it is a speical upc
+            if (specialUPC.Contains(UPC))
+            {
+                amountofDiscount = amountofDiscount +CreateUPCDiscount();
+                return Math.Round(amountofDiscount, 2);
+            }
+            else return Math.Round(amountofDiscount, 2);
+        }
+        double CreateUPCDiscount()
+        {
+            double UPCDiscount =Price * upcDiscount / 100;
+            return UPCDiscount;
         }
 
-        public void Report()
+        void Report()
         {
             if (Discount != 0)
```

Comment:
is the report functionality part of the payment (Price Calculation) service? or it should be a standalone service? 
the class should do only one thing, is your class here doing one only task? read about the Single responsibility principle, please, and let's discuss what you get from it